### DISCUSSION TITLE
BUG: incorrect language detection

### DISCUSSION THREAD

**Discussion by rimutaka** [community/community#22680] [New to GitHub]
<p>It appears that the language detection feature in GitHub is off by a large margin.</p>
<p>My search for recently updated C# projects (<a href="https://api.github.com/search/repositories?q=language:C%23+sort:updated" rel="nofollow noopener">https://api.github.com/search/repositories?q=language:C%23+sort:updated</a>)</p>
<p>returned plenty of non-C# project labeled as C#.</p>
<p>E.g. <a href="https://github.com/haru-shinobu/StickFigure" rel="nofollow noopener">https://github.com/haru-shinobu/StickFigure</a> is labeled as C# 95.4%<br>
ShaderLab 4.6% when it is just a bunch of Unity assets.</p>
<p>I estimate at least 10% of the results in the last few searches were completely off.</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>Hi <a class="mention" href="/rimutaka">@rimutaka</a> thanks for the great observation.</p>
<p>GitHub determines language statistics based on the total size of the files that belong to each language and we use the Linguist open source library to analyze source code files.</p>
<p>I suspect those Unity files are quite large compared to the other language files in that repo.<br>
You can find more information here about <a href="https://github.com/github/linguist#my-repository-is-detected-as-the-wrong-language">why a repository detected the wrong language</a>.</p>
